src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -759,15 +784,21 @@ private int decodeProtocolHeader(
         final int protocolId = protocolHeader.id();
         int progress;
 
-        if (protocolId == 0)
+        switch (protocolId)
         {
+        case DEFAULT_PROTOCOL_ID:
             server.onDecodeProtocolHeader(traceId, authorization, protocolHeader);
             server.decoder = decodeFrame;
-        }
-        else if (protocolId == SASL_PROTOCOL_ID)
-        {
+            break;
+        case TLS_PROTOCOL_ID:
+            // TODO: support TLS
+            break;
```

Comment:
We need to remove this `case`, since otherwise we'd be ignoring `protocol-id` `2`, instead we should be rejecting it, at least for now.